alpha/lib/model/asset.php

```diff
@@ -547,6 +547,22 @@ public function getDownloadUrl($useCdn = false, $forceProxy = false, $preview =
 
 		return $this->getDownloadUrlWithExpiry(86400, $useCdn, $forceProxy, $preview);
 	}
+
+	public function getServeFlavorUrl($entry)

```

Comment:
No need to pass the entry to the asset.
If you need the entry within the asset use $this->getentry().